build.rs

```diff
@@ -552,6 +567,16 @@ fn cc(
     let is_musl = target.env.starts_with("musl");
 
     let mut c = cc::Build::new();
+
+    if target.os == WINDOWS && target.arch == AARCH64 {
+        let _ = c.compiler("clang-cl");
```

Comment:
@awakecoding @briansmith Gentlemen, please review this proposal. This would allow us to avoid Windows AArch64 branching in most cases. The additional flags to disable some warning are due to the fact that clang and clang-cl set different default warnings. BoringSSL sets `-W3` flag for clang-cl instead of `-Wall`, but in our case the `/Wall` flag is part of a static array that I didn't want to refactor, so I just disabled those warnings that prevent the code from compiling.

Reply:
> @awakecoding @briansmith Gentlemen, please review this proposal. This would allow us to avoid Windows AArch64 branching in most cases. The additional flags to disable some warnings are due to the fact that clang and clang-cl set different default warnings. BoringSSL sets `-W3` flag for clang-cl instead of `-Wall`, but in our case the `/Wall` flag is part of a static array that I don't want to refactor, so I just disabled those warnings that prevent the code from compiling. I expect @briansmith's work on supporting clang-cl for all platforms to work around this issue in a better way.

Indeed, I've already prepared a PR that enables compiling x86_64 and i686 Windows with clang and clang-cl, which deals with the warnings, which will obviate (I hope) the need for any such warning-related changes specific to AAarch64.

I am hoping that with my other changes, the user can just set `TARGET_CC=clang` or `TARGET_CC=clang-cl`, and that they'll have the freedom to choose between either clang or clang-cl or whatever other compiler they want (that supports uint128_t), when targeting AAarch64, without having to hard-code anything into build.rs, or where there's as little hard-coded into build.rs as possible.